src/Tickets/Commerce/Cart/Agnostic_Cart.php

```diff
@@ -0,0 +1,525 @@
+	 * Process the items in the cart.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $data to be processed by the cart.
+	 *
+	 * @return array|bool
+	 */
+	public function process( array $data = [] ) {
+		if ( empty( $data ) ) {
+			return false;
+		}
+
+		// Reset the contents of the cart.
+		$this->clear();
+
+		/** @var Tickets_Handler $tickets_handler */
+		$tickets_handler = tribe( 'tickets.handler' );
+
+		// Prepare the error message array.
+		$errors = [];
+
+		// Natively handle adding tickets as items to the cart.
+		foreach ( $data['tickets'] as $ticket ) {
+			// Enforces that the min to add is 1.
+			$quantity = max( 1, (int) $ticket['quantity'] );
+
+			// Check if the ticket can be added to the cart.
+			$can_add_to_cart = $tickets_handler->ticket_has_capacity( $ticket['ticket_id'], $quantity, $ticket['obj'] );
+
+			// Skip and add to the errors if the ticket can't be added to the cart.
+			if ( is_wp_error( $can_add_to_cart ) ) {
+				$errors[] = $can_add_to_cart;
+
+				continue;
+			}
+
+			// Add to / update quantity in cart.
+			$this->add_item( $ticket['ticket_id'], $quantity, $ticket['extra'] );
+		}
+
+		/**
+		 * Fires after the ticket data has been processed.
+		 *
+		 * This allows for further processing of data within the $data array.
+		 *
+		 * @since TBD
+		 *
+		 * @param Cart_Interface $cart The cart object.
+		 * @param array          $data The data to be processed by the cart.
+		 */
+		do_action( 'tec_tickets_commerce_cart_process', $this, $data );
+
+		// Saved added items to the cart.
+		$this->save();
+
+		if ( ! empty( $errors ) ) {
+			return $errors;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Add the full set of parameters to the items in the cart.
+	 *
+	 * @param array $items The items in the cart.
+	 *
+	 * @return array The items in the cart with the full set of parameters.
+	 */
+	protected function add_full_item_params( array $items ): array {
+		return array_map(
+			function ( $item ) {
+				$type = $item['type'] ?? 'ticket';
+				switch ( $type ) {
+					case 'ticket':
+						return $this->add_ticket_params( $item );
+
+					default:
+						/**
+						 * Filter the full item parameters for the cart.
+						 *
+						 * This allows for further processing of the item parameters.
+						 *
+						 * If the item shouldn't be processed, `null` should be returned. Otherwise,
+						 * an array of the full item parameters should be returned.
+						 *
+						 * @since TBD
+						 *
+						 * @param array|null $params The full item parameters for the cart.
+						 * @param array      $item   The item in the cart.
+						 * @param string     $type   The type of item.
+						 */
+						return apply_filters( 'tec_tickets_commerce_cart_add_full_item_params', null, $item, $type );
+				}
+			},
+			$items
+		);
+	}
+
+	/**
+	 * Add the ticket parameters to the item in the cart.
+	 *
+	 * @param array $item The item in the cart.
+	 *
+	 * @return ?array The item in the cart with the full set of parameters.
+	 */
```

Comment:
```suggestion
	/**
	 * Add the ticket parameters to the item in the cart.
	 *
	 * @since TBD
	 *
	 * @param array $item The item in the cart.
	 *
	 * @return ?array The item in the cart with the full set of parameters.
	 */
```